packages/dds/tree/src/test/simple-tree/api/treeApi.spec.ts

```diff
@@ -1301,6 +1302,36 @@ describe("treeNodeApi", () => {
 					});
 				}
 			}
+
+			describe("with misaligned view and stored schema", () => {
+				it("does not preserve additional optional fields", () => {
+					// (because stored keys are not being used, see analogous test in roundtrip-stored)
+					const sf1 = new SchemaFactoryAlpha("com.example");
+					const sf2 = new SchemaFactoryAlpha("com.example");
+					class Point2D extends sf1.object(
+						"Point",
+						{
+							x: sf1.number,
+							y: sf1.number,
+						},
+						{ allowUnknownOptionalFields: true },
+					) {}
+					class Point3D extends sf2.object("Point", {
+						x: sf2.number,
+						y: sf2.number,
+						z: sf2.optional(sf2.number),
+					}) {}
+
+					const testTree = new Point3D({ x: 1, y: 2, z: 3 });
+					const exported = TreeAlpha.exportVerbose(testTree);
+
+					// TODO: The error here should be more clear.
```

Comment:
I'd rather file follow-up work than address this here, but up to reviewer. Leaving this comment for now to track one of the two courses of action.